backend/ingestion/extract_census_data.py

```diff
@@ -1 +1,147 @@
- 
\ No newline at end of file
+import os
+import sys
+import csv
+import logging
+import requests
+from typing import List, Dict
+from dotenv import load_dotenv
+
+########################################################################################
+# SETUP
+########################################################################################
+
+# Enviornment variables
+load_dotenv()
+CENSUS_API_KEY = os.getenv("CENSUS_API_KEY")
+
+# Logging configuration
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+# Mappings
+variable_ids = {"B01001_001E": "total_population"}
+city_fips = {
+    "nyc": {"state_fips": "36", "county_fips": ["061", "047", "081", "005", "085"]},
+    "chicago": {"state_fips": "17", "county_fips": ["031"]},
+    "portland": {"state_fips": "41", "county_fips": ["051"]},
+}
+
+########################################################################################
+# HELPER FUNCTIONS
+########################################################################################
+
+
+def valid_command_line_arg(supported_cities: List[str]) -> str:
+    """
+    Validates command line use and extracts argument.
+
+    Inputs:
+        available_cities (List[str]): List of cities that are valid inputs.
+
+    Returns:
+      A string referring to the city to pull data from.
+    """
+    # Check syntax
+    if len(sys.argv) != 2:
+        sys.exit("Retype command as 'python3 extract_census_data.py <city_name>'")
+    # Check city
+    city = sys.argv[1]
+    if city not in supported_cities:
+        sys.exit(
+            "Unsupported city." + f" Available options: {', '.join(supported_cities)}"
+        )
+    return city
+
+
+def get_census_data(
+    variable_ids: Dict[str, str], state_code: str, county_code: str
+) -> List[Dict]:
+    """
+    Fetches data from US Census API.
+
+    Inputs:
+        variable_ids (dict): Unique variable IDs and their names
+        state_code (str): State-level FIPS code
+        county_code (str): County-level FIPS code
+
+    Returns:
+        A list of dictionaries with function inputs as key, value pairs.
+    """
+    # Request parameters
+    url = "https://api.census.gov/data/2022/acs/acs5"
```

Comment:
Are we pulling everything from this ACS survey?, maybe we could add endpoint to the function parameters and leave the 2022 acs as default.
If we are expecting to only use 2022 this seems good